co_sim_io/c/co_sim_io_c_model_part.h

```diff
@@ -55,24 +55,32 @@ typedef enum
     CoSimIO_Point3D
 } CoSimIO_ElementType;
 
+
+// Node functions
 int CoSimIO_Node_Id(CoSimIO_Node I_Node);
 double CoSimIO_Node_X(CoSimIO_Node I_Node);
 double CoSimIO_Node_Y(CoSimIO_Node I_Node);
 double CoSimIO_Node_Z(CoSimIO_Node I_Node);
+// double* CoSimIO_Coordinates(CoSimIO_Node I_Node);
 
 
+// Element functions
 int CoSimIO_Element_Id(CoSimIO_Element I_Element);
 CoSimIO_ElementType CoSimIO_Element_Type(CoSimIO_Element I_Element);
 int CoSimIO_Element_NumberOfNodes(CoSimIO_Element I_Element);
+// CoSimIO_Node* CoSimIO_Element_Nodes(CoSimIO_Element I_Element);
 
 
+// ModelPart functions
 CoSimIO_ModelPart CoSimIO_CreateModelPart(const char* I_Name);
 
 int CoSimIO_FreeModelPart(CoSimIO_ModelPart I_ModelPart);
 
 const char* CoSimIO_ModelPart_Name(CoSimIO_ModelPart I_ModelPart);
 int CoSimIO_ModelPart_NumberOfNodes(CoSimIO_ModelPart I_ModelPart);
 int CoSimIO_ModelPart_NumberOfElements(CoSimIO_ModelPart I_ModelPart);
+// CoSimIO_Node* CoSimIO_ModelPart_Nodes(CoSimIO_ModelPart I_ModelPart);
+// CoSimIO_Element* CoSimIO_ModelPart_Elements(CoSimIO_ModelPart I_ModelPart);
```

Comment:
@pooyan-dadvand this is where I am not sure what would be better

Reply:
This is correct. Arrays in c are returned by pointer to the first element. Using this one can use normal indexing our pointer jump to access to the members of the array. 

What is important is the memory management that here everything would belong to the CosimIO. Means that you should have this array also internally stored and do not create it inside your function.

Reply:
I think that maybe a cleaner solution would be to provide only index based access to the entities like:
```
CoSimIO_Node* CoSimIO_ModelPart_Node_By_Index(CoSimIO_ModelPart I_ModelPart, int I_node_index);
```